docs/source/preparing_datasets/basic_dataset_conversion.md

```diff
@@ -0,0 +1,330 @@
+| ---------- | ------------ |
+| 'xxh32'    | 4            |
+| 'xxh64'    | 8            |
+| 'xxh128'   | 16           |
+| 'xxh3_64'  | 8            |
+| 'xxh3_128' | 16           |
+
+As an example:
+
+<!--pytest.mark.skip-->
+```python
+hashes = ['sha256', 'xxh64']
+```
+
+## Example: Writing a dataset to MDS format
+
+Let's put it all together with an example. Here, we create a synthetic classification dataset that returns a tuple of features and a label.
+
+```python
+import numpy as np
+
+class RandomClassificationDataset:
+    """Classification dataset drawn from a normal distribution.
+
+    Args:
+        shape: data sample dimensions (default: (10,))
+        size: number of samples (default: 10000)
+        num_classes: number of classes (default: 2)
+    """
+
+    def __init__(self, shape=(10,), size=10000, num_classes=2):
+        self.size = size
+        self.x = np.random.randn(size, *shape)
+        self.y = np.random.randint(0, num_classes, size=(size,)).tolist()
+
+    def __len__(self):
+        return self.size
+
+    def __getitem__(self, index: int):
+        return self.x[index], self.y[index]
+```
+
+Here, we write shards to a local directory. You can specify a remote path as well.
+<!--pytest-codeblocks:cont-->
+```python
+output_dir = 'test_output_dir'
+```
+
+Specify the column encoding types for each sample and label:
+<!--pytest-codeblocks:cont-->
+```python
+columns = {'x': 'pkl', 'y': 'int'}
+```
+
+Optionally, specify a compression algorithm and level:
+<!--pytest-codeblocks:cont-->
+```python
+compression = 'zstd:7' # compress shards with ZStandard, level 7
+```
+
+Optionally, specify a list of hash algorithms for verification:
+<!--pytest-codeblocks:cont-->
+```python
+hashes = ['sha1'] # Use only SHA1 hashing on each shard
+```
+
+Optionally, provide a shard size limit, after which a new shard starts. In this small example, we use 10kb, but for production datasets 50-100MB is more appropriate.
+<!--pytest-codeblocks:cont-->
+```python
+# Here we use a human-readable string, but we could also
+# pass in an int specifying the number of bytes.
+limit = '10kb'
+```
+
+The last thing we need is a generator that iterates over the raw data samples.
+<!--pytest-codeblocks:cont-->
+```python
+def each(samples):
+    """Generator over raw dataset samples.
+
+    Args:
+        samples: Raw samples as tuples of (feature, label).
+
+    Yields:
+        Each sample, as a dict.
+    """
+    for x, y in samples:
+        yield {
+            'x': x,
+            'y': y,
+        }
+```
+
+It's time to call the {class}`streaming.MDSWriter` with the above initialized parameters and write the samples by iterating over a dataset.
+<!--pytest-codeblocks:cont-->
+```python
+from streaming.base import MDSWriter
+
+dataset = RandomClassificationDataset()
+with MDSWriter(out=output_dir, columns=columns, compression=compression, hashes=hashes, size_limit=limit) as out:
+    for sample in each(dataset):
```

Comment:
can we make this more direct, and drop the "each"? i think each is a functional-ism for when things get more complicated, but here that's not what the point that we're trying to draw attention to...

```
for x, y in dataset:
    out.write({'x': x, 'y': y})
```

Reply:
ooh true, thanks